src/opm/parser/eclipse/EclipseState/Aquancon.cpp

```diff
@@ -80,20 +81,36 @@ namespace Opm {
                 for (int j=m_aqurecord.at(aquanconRecordIdx).j1; j <= m_aqurecord.at(aquanconRecordIdx).j2; j++)
                     for (int i=m_aqurecord.at(aquanconRecordIdx).i1; i <= m_aqurecord.at(aquanconRecordIdx).i2; i++)
                         m_aqurecord.at(aquanconRecordIdx).global_index_per_record.push_back
-                                                            (grid.getGlobalIndex(i-1 ,j-1 ,k- 1)
+                                                            (
+                                                                grid.getGlobalIndex(i-1, j-1, k-1)
                                                             );
             }
             size_t global_index_per_record_size = m_aqurecord.at(aquanconRecordIdx).global_index_per_record.size();
-            m_aqurecord.at(aquanconRecordIdx).influx_coeff_per_record.resize(global_index_per_record_size,influx_coeff);
+
+            m_aqurecord.at(aquanconRecordIdx).influx_coeff_per_record.resize(global_index_per_record_size, nullptr);
+
+            if (aquanconRecord.getItem("INFLUX_COEFF").hasValue(0))
+            {
+                double* influx_coeff = new double( aquanconRecord.getItem("INFLUX_COEFF").getSIDouble(0) );
+
+                std::for_each(
```

Comment:
Simpler to use a range-for (assuming @bska's fix is applied above):
```c++
for (auto& influx : m_aqurecord.at(aquanconRecordIdx).influx_coeff_per_record) {
    influx.reset(new double(influx_coeff));
}
```

Reply:
Yes indeed. Makes it much cleaner. Thanks for the comment.